storage/engine/mvcc.go

```diff
@@ -835,11 +835,7 @@ func MVCCIncrement(engine Engine, ms *MVCCStats, key proto.Key, timestamp proto.
 // containing the actual value.
 func MVCCConditionalPut(engine Engine, ms *MVCCStats, key proto.Key, timestamp proto.Timestamp, value proto.Value,
 	expValue *proto.Value, txn *proto.Transaction) error {
-	// Handle check for non-existence of key. In order to detect
-	// the potential write intent by another concurrent transaction
-	// with a newer timestamp, we need to use the max timestamp
-	// while reading.
-	existVal, _, err := MVCCGet(engine, key, proto.MaxTimestamp, true /* consistent */, txn)
+	existVal, _, err := MVCCGet(engine, key, timestamp, true /* consistent */, txn)
```

Comment:
Can you add a comment here explaining that it's ok to read at `timestamp`? A naive look would suspect that there might be a bug here.